### PR TITLE
Zendesk BatchSource Plugin

### PR DESCRIPTION
**Pull request by amolkokane-cirus**
Removed MultiSource plugin as it is not working
Added missing javadocs
Changed copyright to 2020

### CODE REVIEW

File: src/main/java/io/cdap/plugin/zendesk/source/batch/http/HttpUtil.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      authCache.put(targetHost, new BasicScheme());
+    }
+    HttpClientContext context = HttpClientContext.create();
+    context.setCredentialsProvider(credentialsProvider);
+    context.setAuthCache(authCache);
+    return context;
+  }
+
+  /**
+   * Creates a ulr for the first page.
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/zendesk/source/batch/http/HttpUtil.java]: nit: ulr->url

File: src/main/java/io/cdap/plugin/zendesk/source/batch/http/PagedIterator.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    if (!objectType.isBatch()) {
+      return (String) responseMap.get(NEXT_PAGE);
+    }
+
+    String next = (String) responseMap.get(NEXT_PAGE);
+    if (next == null) {
+      return null;
+    }
+
+    // A request after five minutes ago will result in a 422 responds from Zendesk.
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/zendesk/source/batch/http/PagedIterator.java]: nit: responds -> response

File: src/main/java/io/cdap/plugin/zendesk/source/common/config/BaseZendeskSourceConfig.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+  @Name(PROPERTY_SUBDOMAINS)
+  @Description("Zendesk Subdomains to read objects from.")
+  @Macro
+  private String subdomains;
+
+  @Name(PROPERTY_OBJECTS_TO_PULL)
+  @Description("Objects to pull from Zendesk API.")
+  @Nullable
+  private String objectsToPull;
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/zendesk/source/common/config/BaseZendeskSourceConfig.java:85]: should this be macro enabled? In other places I see code that checks `config.containsMacro(BaseZendeskSourceConfig.PROPERTY_OBJECTS_TO_PULL)`

File: src/main/java/io/cdap/plugin/zendesk/source/batch/http/HttpUtil.java (hunk 1 above)
Comment (amolkokane-cirus) [reply src/main/java/io/cdap/plugin/zendesk/source/batch/http/HttpUtil.java]: Done

File: src/main/java/io/cdap/plugin/zendesk/source/batch/http/PagedIterator.java (hunk 2 above)
Comment (amolkokane-cirus) [reply src/main/java/io/cdap/plugin/zendesk/source/batch/http/PagedIterator.java]: Done

File: src/main/java/io/cdap/plugin/zendesk/source/common/config/BaseZendeskSourceConfig.java (hunk 3 above)
Comment (amolkokane-cirus) [reply src/main/java/io/cdap/plugin/zendesk/source/common/config/BaseZendeskSourceConfig.java:85]: Done

**Review by rmstar — APPROVED** [review]

**Pull request merged**